database/src/main/java/com/firebase/ui/database/FirebaseArray.java

```diff
@@ -97,6 +97,7 @@ public void onChildMoved(DataSnapshot snapshot, String previousChildKey) {
         mSnapshots.remove(oldIndex);
         int newIndex = previousChildKey == null ? 0 : (getIndexForKey(previousChildKey) + 1);
         mSnapshots.add(newIndex, snapshot);
+        mSnapshotMap.put(snapshot.getKey(), newIndex);
```

Comment:
I don't think we have any tests that actually cover item re-order events, we should add one if we are going to change this functionality.